controllers/operatorpolicy_controller.go

```diff
@@ -1440,6 +1445,128 @@ func (r *OperatorPolicyReconciler) musthaveSubscription(
+	}
+
+	// Extract package name
+	packageName, ok, err := unstructured.NestedString(packageManifest.Object, "status", "packageName")
+	if err != nil || !ok {
+		return fmt.Errorf("failed to get packageName in the pakcageManifest %q %w", desiredSub.Spec.Package, err)
+	}
+
+	// Check if the package is deprecated
+	if deprecation, ok, _ := unstructured.NestedMap(packageManifest.Object, "status", "deprecation"); ok {
+		message, _, _ := unstructured.NestedString(deprecation, "message")
+		updateStatus(policy, deprecationCond(packageName, Package, message))
+
+		return nil
+	}
+
+	// Check the selected channel is deprecated
+	selectedChannelName := desiredSub.Spec.Channel
+	if selectedChannelName == "" {
+		defaultChannel, ok, err := unstructured.NestedString(packageManifest.Object, "status", "defaultChannel")
+		if err != nil || !ok {
+			return fmt.Errorf("failed to retrieve default channel in PackageManifest %q: %w",
+				desiredSub.Spec.Package, err)
+		}
+
+		selectedChannelName = defaultChannel
+	}
+
+	// Extract channels
+	channels, ok, err := unstructured.NestedSlice(packageManifest.Object, "status", "channels")
+	if err != nil || !ok {
+		return fmt.Errorf("failed to retrieve channels in PackageManifest %q: %w", desiredSub.Spec.Package, err)
+	}
+
+	// Check for deprecation in the selected channel
+	for _, channel := range channels {
+		channelMap, ok := channel.(map[string]interface{})
+		if !ok {
+			return fmt.Errorf("failed to parse channel in PackageManifest %q: %w", desiredSub.Spec.Package, err)
+		}
+
+		channelName, ok := channelMap["name"].(string)
+		if !ok || channelName != selectedChannelName {
+			continue
+		}
+
+		// Check if the channel is deprecated
+		if channelDep, ok := channelMap["deprecation"].(map[string]interface{}); ok {
+			message, _, _ := unstructured.NestedString(channelDep, "message")
+			updateStatus(policy, deprecationCond(channelName, Channel, message))
+
+			return nil
+		}
+
+		// Check bundle Deprecation
+		if csv == nil {
+			return nil
+		}
+
+		csvName := csv.GetName()
+
+		entries, ok, err := unstructured.NestedSlice(channelMap, "entries")
+		if err != nil || !ok {
+			return fmt.Errorf("failed to parse entries in channel %q of PackageManifest %q: %w",
+				channelName, desiredSub.Spec.Package, err)
+		}
+
+		for i, entry := range entries {
+			entryMap, ok := entry.(map[string]interface{})
+			if !ok {
+				return fmt.Errorf(
+					"failed to parse entry index [%d] in channel %q of PackageManifest %q: %w",
+					i, channelName, desiredSub.Spec.Package, err,
+				)
+			}
+
+			entryName, _, _ := unstructured.NestedString(entryMap, "name")
+			if entryName == csvName {
+				if bundleDeprecation, ok := entryMap["deprecation"].(map[string]interface{}); ok {
+					message, _, _ := unstructured.NestedString(bundleDeprecation, "message")
+					updateStatus(policy, deprecationCond(entryName, Bundle, message))
+
+					return nil
+				}
+			}
+		}
+	}
+
+	return nil
```

Comment:
I think it should update the deprecation condition here to reflect that nothing deprecated was found. Otherwise, if I have a deprecated version, then I update to something supported, I think the OperatorPolicy will continue to report the (stale) deprecation.